packages/zendesk-adapter/src/filters/guide_order/guide_orders_utils.ts

```diff
@@ -0,0 +1,155 @@
+  [BRAND_TYPE_NAME]: () =>
+    new ObjectType({
+      elemID: new ElemID(ZENDESK, ORDER_IN_BRAND_TYPE),
+      fields: { [CATEGORIES_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) } },
+    }),
+  [CATEGORY_TYPE_NAME]: () =>
+    new ObjectType({
+      elemID: new ElemID(ZENDESK, ORDER_IN_CATEGORY_TYPE),
+      fields: { [SECTIONS_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) } },
+    }),
+  [SECTION_TYPE_NAME]: () =>
+    new ObjectType({
+      elemID: new ElemID(ZENDESK, ORDER_IN_SECTION_TYPE),
+      fields: {
+        [SECTIONS_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) },
+        [ARTICLES_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) },
+      },
+    }),
+}
+
+export const createOrderType = (typeName: string) : ObjectType => GUIDE_ORDER_TYPES[typeName]()
+
+
+export const createOrderElement = ({ parent, parentField, orderField, childrenElements, orderType }
+: {
+  parent: InstanceElement
+  parentField: string
+  orderField: string
+  childrenElements: InstanceElement[]
+  orderType: ObjectType
+}): InstanceElement => {
+  const parentsChildren = _.orderBy(
+    childrenElements.filter(c => c.value[parentField] === parent.value.id),
+    // Lowest position index first, if there is a tie - the newer is first, another tie - by id
+    ['value.position', 'value.created_at', 'value.id'], ['asc', 'desc', 'asc']
+  )
+
+  return new InstanceElement(
+    `${parent.elemID.name}_${orderField}`,
+    orderType,
+    {
+      [orderField]: parentsChildren.map(c => new ReferenceExpression(c.elemID, c)),
+    },
+    // The same directory as it's parent
+    parent.path && [...parent.path.slice(0, -1), `${orderField}_order`],
+    {
+      [CORE_ANNOTATIONS.PARENT]: [new ReferenceExpression(parent.elemID, parent)],
+    }
+  )
+}
+
+// Transform order changes to new changes and deploy them
+export const deployOrderChanges = async ({ changes, client, config, orderField } : {
+  changes: Change<InstanceElement>[]
+  client: ZendeskClient
+  config: FilterContext
+  orderField: string
+}) : Promise<DeployResult> => {
+  const orderChangeErrors: Error[] = []
+  const appliedChanges: Change[] = []
+
+  await awu(changes).forEach(async change => {
+    // Removal means nothing because the element is internal
+    // We have order_deletion_validator to make sure the parent was also deleted
+    if (isRemovalChange(change)) {
+      appliedChanges.push(change)
+      return
+    }
+
+    const newOrderElement = change.data.after
+    const orderElements = newOrderElement.value[orderField] || []
+
+    await awu(orderElements).filter(isReferenceExpression).forEach(async (child, i) => {
+      const resolvedChild = child.value
+      const childType = resolvedChild.elemID.typeName
+      const childUpdateApi = config[API_DEFINITIONS_CONFIG].types[childType].deployRequests?.modify
+
+      // Send an api request to update the positions of the elements in the order list
+      if (childUpdateApi !== undefined) {
```

Comment:
nit: if we expect to have this config, it's usually better to use
```suggestion
      if (childUpdateApi === undefined) {
        ...
        return
      }
```
and avoid the extra nesting level for the rest of the code